### DISCUSSION TITLE
PhotoSync & PhotoPrism - Photos Not Deleting upon import via WebDAV

### DISCUSSION THREAD

**Discussion by chrisatl320** [photoprism/photoprism#3353] [Q&A] [answered]
Hi, 

I have been using PhotoSync for a while to upload my iPhone directly to a share on my Unraid server. However, I hadn't found a great way to catalog the photos and decided to try Photoprism. 

My understanding is the only way for automatic import of my uploads is to use WebDAV. I tried WebDAV last night and drop files into my Import Folder that I designated via Docker within Unraid. 

The photos / videos were imported into the correct dated folder and are showing within the UI. However, the original files are still showing in the import folder. 

It appears that PhotoPrism has renamed the files (which is fine with me) but I can't get the files within the Import folder to automatically delete. Since they've changed names, I'm hesitant to manually delete them since I won't be able to confirm in the future if they have in fact been imported. 

Am I missing something? 

Thanks, 

CH

**Comment by graciousgrey** [comment, marked as answer]
Hi :) 

If you enable the "Move" option on the import page. Files will be removed from the import folder once they have been added to Originals or if they already exist in Originals. 

   https://docs.photoprism.app/user-guide/library/import/#when-should-move-files-be-selected

**Comment by chrisatl320** [comment]
Perfect! Thanks so much. I'll give this a try and see how it works.

**Comment by hypnotoad** [comment]
I have the feeling that it is not working as expected. When I manually import (move option is always selected), the source file gets deleted. The automatic import via webdav does not seem to delete the source images.

By this, I suspect that files get imported again after I deleted them (because they remain in the import folder).

**Reply by graciousgrey** [reply]
I just tested this and it works as expected on our setups: If I add photos to the import folder using WebDAV and the move option is selected the import that is triggered automatically after the safety delay moves the files from import to originals while if the move option is not selected the import only copies the files. 

Are you using our latest release? Is it possible that the files that remain in the import folder even if "move" is selected are unsupported?

**Reply by hypnotoad** [reply]
I use release 231128-f48ff16ef which is probably not the latest but newer than the previous replies in this thread. It is about jpeg files. Here is the log of a file that I just uploaded via webdav and which was not deleted so far:
```
2024-02-29 18:09:11 INFO import completed in 13 s
2024-02-29 18:08:59 INFO import: added main jpg file 2024/02/20240229_170540_91061A2D.jpg
2024-02-29 18:08:59 ERRO photo: maps: location lookup disabled (find location)
2024-02-29 18:08:59 INFO media: 20240229_170540_91061A2D.jpg was taken at 2024-02-29 17:05:40 +0000 UTC (meta)
2024-02-29 18:08:59 INFO media: created 11 thumbnails for 2024/02/20240229_170540_91061A2D.jpg [1.342001162s]
2024-02-29 18:08:57 INFO import: moving main jpg file IMG_20240229_180540_259.jpg to 2024/02/20240229_170540_91061A2D.jpg
2024-02-29 18:08:57 INFO media: IMG_20240229_180540_259.jpg was taken at 2024-02-29 17:05:40 +0000 UTC (meta)
2024-02-29 18:08:57 INFO import: found no .ppignore file
2024-02-29 18:08:57 INFO copying files from import
2024-02-29 18:06:04 INFO webdav: PUT /import/IMG_20240229_180540_259.jpg
```

Are there maybe two different move options (one for manual, one for automatic import)? I just know this here which is activated:
![grafik](https://github.com/photoprism/photoprism/assets/7377727/93444935-6805-44e1-b988-61abad42e115)

**Reply by graciousgrey** [reply]
The logs looks good to me. 

Just to be sure, did you open the import folder to confirm that the file is still there after the import?

**Reply by hypnotoad** [reply]
Yes, it is there after the automatic webdav import.
```
$ ls -l
insgesamt 5129
-rw-rw-r-- 1 root root 4239316 29. Feb 18:06 IMG_20240229_180540_259.jpg
```

It gets removed after I did a manual import
```
2024-02-29 19:11:49 INFO import completed in 11 s
2024-02-29 19:11:38 INFO import: imported 2 files
2024-02-29 19:11:38 INFO import: deleted IMG_20240229_180540_259.jpg (already exists)
2024-02-29 19:11:38 INFO import: IMG_20240229_180540_259.jpg is identical to 2024/02/20240229_170540_91061A2D.jpg (sha1 49522cad1b9e681dda6c2f1182bb3c143d4ce9e2)
2024-02-29 19:11:38 INFO media: IMG_20240229_180540_259.jpg was taken at 2024-02-29 17:05:40 +0000 UTC (meta)
2024-02-29 19:11:38 INFO import: found no .ppignore file
2024-02-29 19:11:38 INFO moving files from import
```

```
$ ls -l
insgesamt 0
```

**Reply by hypnotoad** [reply]
The log lines above differ differ in moving/copying "files from import". Would it make sense if I test with some newer release?

**Reply by graciousgrey** [reply]
I tested with the same version and it worked for me, so the version is probably not the problem.

**Reply by hypnotoad** [reply]
I saw that my settings.yml has Import.Move=false, while I have a checkmark for moving files in the web interface (cf. above).

My guess is that there are two different "move" settings, one for manual, one for automatic. The documentation seems outdated (e.g. I don't have a library tab), could it be that this changed recently and that there is no "move" setting yet for the autoimport?

**Reply by lastzero** [reply]
Do you use multiple user accounts or could there be filesystem permission issues, so the file cannot be updated?

**Reply by hypnotoad** [reply]
I can confirm that manually changing settings.yml fixed the problem for me.

Regarding permissions: settings.yml indeed was not modified since I installed photoprism in December. But it has the same permissions as hub.yml which was updated recently.

Indeed I created a second user account with Admin role [as documented](https://docs.photoprism.app/user-guide/users/cli/). I used this account for the auto-import and the manual import.

**Reply by lastzero** [reply]
Note that only settings performed by a SUPER ADMIN account will be applied as global defaults:
- https://docs.photoprism.app/user-guide/users/roles/#admin

**Reply by hypnotoad** [reply]
That would explain most of it, thank you. But being new to photoprism, I find it misleading that manual import moving is configured in my user configuration and automatic import moving is configured as super admin for all users.

This should not be a complaint (my problem is solved now) but rather a feedback to make photoprism better.

**Reply by lastzero** [reply]
Thank you for your feedback! We are constantly working to improve our documentation, but also have to spend a lot of our time on developing new features and answering questions from our users.
